sdk/forms/src/main/java/com/klaviyo/forms/KlaviyoWebViewDelegate.kt

```diff
@@ -0,0 +1,244 @@
+package com.klaviyo.forms
+
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebResourceRequest
+import android.webkit.WebResourceResponse
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.core.content.ContextCompat.startActivity
+import androidx.webkit.JavaScriptReplyProxy
+import androidx.webkit.WebMessageCompat
+import androidx.webkit.WebViewCompat
+import androidx.webkit.WebViewFeature.WEB_MESSAGE_LISTENER
+import com.klaviyo.analytics.Klaviyo
+import com.klaviyo.analytics.networking.ApiClient
+import com.klaviyo.core.Registry
+import com.klaviyo.core.config.Clock
+import com.klaviyo.core.lifecycle.ActivityEvent
+import com.klaviyo.core.utils.WeakReferenceDelegate
+import com.klaviyo.forms.BridgeMessage.Companion.handShakeData
+import java.io.BufferedReader
+
+/**
+ * Manages [KlaviyoWebView] to power in-app forms
+ */
+internal class KlaviyoWebViewDelegate : WebViewClient(), WebViewCompat.WebMessageListener {
+    /**
+     * Defines origin(s) for which this delegate should be used
+     */
+    val allowedOrigin = setOf(Registry.config.baseUrl)
+
+    val bridgeName = "KlaviyoNativeBridge"
+
+    private val activity: Activity? get() = Registry.lifecycleMonitor.currentActivity
+
+    /**
+     * For timeout on native bridge "handshake" event
+     */
+    private var handshakeTimer: Clock.Cancellable? = null
+
+    /**
+     * Weak reference to the WebView to avoid memory leak
+     */
+    private var webView: WebView? by WeakReferenceDelegate()
+
+    /**
+     * For tracking device rotation
+     */
+    private var orientation: Int? = null
+
+    init {
+        /**
+         * This closes the form on rotation, which we can detect with the local field
+         * We wait for a change, see if it's different from the current, and close an open webview
+         */
+        Registry.lifecycleMonitor.onActivityEvent {
+            if (it is ActivityEvent.ConfigurationChanged) {
+                val newOrientation = it.newConfig.orientation
+                if (orientation != newOrientation) {
+                    Registry.log.debug("New screen orientation, closing form")
+                    close()
+                }
+                orientation = newOrientation
+            }
+        }
+    }
+
+    /**
+     * Initialize a webview instance, with protection against duplication
+     * and initialize klaviyo.js for in-app forms with handshake data injected in the document head
+     */
+    fun initializeWebView() = webView ?: KlaviyoWebView().also { webView ->
+        this.webView = webView
+
+        val klaviyoJsUrl = Uri.parse(Registry.config.baseCdnUrl)
+            .buildUpon()
+            .path("onsite/js/klaviyo.js")
```

Comment:
the biggest of all nits: make this a const? Similar to how we treat paths in our request classes as part of the analytics module

Reply:
Thats fair!